### ISSUE TITLE
Push Monitor : Return to UP but still DOWN

### ISSUE THREAD

**Issue by Apashh** [louislam/uptime-kuma#1732]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

Hello,

When the probes in inverted mode are called, it goes back to UP after the defined time interval when it should have remained DOWN

![image](https://user-images.githubusercontent.com/83477029/172553514-0b46556c-5d8f-402f-a7e5-d7add95b163c.png)

### 👟 Reproduction steps

1. Configure PUSH Monitor with Inverted Mod
2. Configure interval to : 70 second
3. Call the URL every 60 second

### 👀 Expected behavior

In normal state it should remain DOWN

### 😓 Actual Behavior

It goes UP saying there is no heartbeat in the time window.

### 🐻 Uptime-Kuma Version

1.16.1

### 💻 Operating System and Arch

Debian 11 x64

### 🌐 Browser

Google Chrome

**Comment by kaysond** [comment]
This is probably my fault sorry!! I honestly just never even thought to test inverted push monitors. I'm curious - what are you using it for? 

I haven't had a chance to check, but the problem is probably this conditional which needs to also be inverted
https://github.com/louislam/uptime-kuma/blob/69273a6c41662c3fffea21879b6a8cb6bf642e76/server/model/monitor.js#L392

**Comment by Apashh** [comment]
NP ! 😊
I use this option to check if a disk is down.
I have created a script to verify with SMART if a disk is OK. If not, I push API URL to avert UPTIME that my service is DOWN and if UPTIME don't receive a PUSH (70 seconds), it means my service (disk) is UP. 😊
I hope it's understandable!

**Comment by Apashh** [comment]
I don't understand your line but why before 1.16.1 or 1.16.0 the push monitor worked correctly ?

**Comment by kaysond** [comment]
> I use this option to check if a disk is down.
> I have created a script to verify with SMART if a disk is OK. If not, I push API URL to avert UPTIME that my service is DOWN and if UPTIME don't receive a PUSH (70 seconds), it means my service (disk) is UP. 😊
> I hope it's understandable!

Ah I understand. I would recommend doing it the other way: if the disk is OK, you should push to keep it UP. 
The way you have it now, if your script stops running, the check will stay UP even though its not running at all!

> I don't understand your line but why before 1.16.1 or 1.16.0 the push monitor worked correctly ? 

For 1.16.1 I submitted a PR that fixed a race condition in the push monitors which added that line, but it breaks inverted monitors!

**Comment by Apashh** [comment]
Ok thank you !
What can I do to fix it before the new version ? 
Because I have many test like "pour-cent disk use" "process use (%)" etc, which force me to use the inverted mod..

**Comment by kaysond** [comment]
> What can I do to fix it before the new version ?

I can't think of a workaround. You might be able to revert to 1.16.0, but I would back up everything first because the DB is slightly different.

> Because I have many test like "pour-cent disk use" "process use (%)" etc, which force me to use the inverted mod..

Not sure what your exact setup is, but you could just flip all of your tests. e.g.

```
if (% disk use < limit)
  push UP

if (% process use < limit)
  push UP

if (disk ok)
  push UP
```

**Comment by Apashh** [comment]
Oh ok... I will test to revert in 1.16.0! 

Yeah I know, my script allows if my equipment is down to give the exact percentage of use..
While with this new technique I will have a message that when it is ok with message and when it exceeds the limit I will not know by how much in percent.. 

You think it's possible it will fix in the new version ? 👌😉

**Comment by Apashh** [comment]
Ok, thanks for your reply! I hope you will check today! 

feel free to ask me questions if needed! thanks again

**Comment by kaysond** [comment]
@Apashh fix submitted (#1762). hopefully it can get in for 1.17.0
